File: app/src/androidTest/java/cn/jinelei/jyhome/page/home/MainActivityTest.java
package cn.jinelei.jyhome.page.home;


import androidx.test.espresso.assertion.ViewAssertions;
import androidx.test.ext.junit.rules.ActivityScenarioRule;
import androidx.test.filters.LargeTest;
import androidx.test.runner.AndroidJUnit4;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import cn.jinelei.jyhome.R;
import cn.jinelei.jyhome.page.MainActivity;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withId;

@RunWith(AndroidJUnit4.class)
@LargeTest
public class MainActivityTest {
    @Rule
    public ActivityScenarioRule<MainActivity> activityScenarioRule
            = new ActivityScenarioRule<>(MainActivity.class);

    @Test
    public void testBtnFailure() {
        onView(withId(R.id.btn_failure)).perform(click());
        onView(withId(R.id.pb_loading)).check(ViewAssertions.matches(isDisplayed()));
    }

}

File: app/src/androidTest/java/cn/jinelei/jyhome/page/main/home/MainActivityTest.kt
package cn.jinelei.jyhome.page.main.home

import androidx.lifecycle.ViewModel
import androidx.test.core.app.launchActivity
import androidx.test.ext.junit.runners.AndroidJUnit4
import cn.jinelei.jyhome.page.main.MainActivity
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class MainActivityTest {
    @Test
    fun testEvent() {
        val scenario = launchActivity<MainActivity>()
//        onView(withId(R.id.btn_success)).perform(click())
//                .check(onView(withId(R.id.pb_loading)).check(matches(isDisplayed()))
    }
}